pkg/controller/vsphere/machine_scope.go

```diff
@@ -74,8 +74,10 @@ func newMachineScope(params machineScopeParams) (*machineScope, error) {
 	if providerSpec.Workspace == nil {
 		return nil, fmt.Errorf("%v: no workspace provided", params.machine.GetName())
 	}
-	authSession, err := session.GetOrCreate(context.TODO(),
-		providerSpec.Workspace.Server, providerSpec.Workspace.Datacenter,
+
+	server := fmt.Sprintf("%s:%s", providerSpec.Workspace.Server, getPortFromConfig(vSphereConfig))
+	authSession, err := session.GetOrCreate(params.Context,
+		server, providerSpec.Workspace.Datacenter,
```

Comment:
if no port is set, this would pass `1.1.1.1:` and use that to create the session key. Probably not harmful, but odd at least.

Reply:
Yes, having `:` at the end of URL is not harmful and that's I left it as it is.